testsuite/MDAnalysisTests/lib/test_autocorrelation.py

```diff
@@ -0,0 +1,74 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver Beckstein
+# and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from MDAnalysis.lib import autocorrelation
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal, assert_
+from six.moves import range
+
+
+
+class TestAutoCorrelation(object):
+    @staticmethod
+    def reference(vecs):
```

Comment:
Have a look at the testsuite of statsmodels. They implemented a very well tested autocorrelation function.